posix/iov.c

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/safecrt.h>
+#include <openenclave/corelibc/limits.h>
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/sys/uio.h>
+#include <openenclave/internal/posix/iov.h>
+
+/* Get the deflated size of the IO vector. */
+size_t oe_iov_compute_size(const struct oe_iovec* iov, size_t iov_count)
+{
+    size_t size = 0;
+
+    for (size_t i = 0; i < iov_count; i++)
+        size += iov[i].iov_len;
```

Comment:
>size += iov[i].iov_len; [](start = 8, length = 23)

Good enough for now, consider filing an issue to track switching to EDL with nested type marshalling, or add integer overflow checks.

Reply:
added overflow checking with safemath.h